### PR TITLE
Add support for image mutation

### LINKED ISSUE #54: Support image mutation datatype

### PR DESCRIPTION
**Pull request by dvesh3**
## Changes in this pull request  
Resolves #54

### CODE REVIEW

File: src/GraphQL/MutationFieldConfigGenerator/Image.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+     */
+    public function getGraphQlMutationFieldConfig($nodeDef, $class, $container = null)
+    {
+        $processor = new \Pimcore\Bundle\DataHubBundle\GraphQL\InputProcessor\Image($nodeDef);
+        $processor->setGraphQLService($this->getGraphQlService());
+
+        return [
+            'arg' => $this->getGraphQlService()->getTypeDefinition("image_input"),
+            'processor' => [$processor, 'process'],
+            'description' => "Either as Asset ID or Asset Path"
```
Comment (weisswurstkanone) [inline src/GraphQL/MutationFieldConfigGenerator/Image.php]: Is this still valid?

File: src/GraphQL/Type/InputType/AbstractRelationInputType.php [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+    /**
+     * @param array $config
+     * @param array $context
+     */
+    public function build(&$config)
+    {
+        $config['fields'] = [
+            'id' => [
+                'type' => Type::int()
```
Comment (weisswurstkanone) [inline src/GraphQL/Type/InputType/AbstractRelationInputType.php]: should be mandatory I guess

File: src/GraphQL/InputProcessor/Image.php [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+     * @throws \Exception
+     */
+    public function process(Concrete $object, $newValue, $args, $context, ResolveInfo $info)
+    {
+        $attribute = $this->getAttribute();
+        Service::setValue($object, $attribute, function($container, $setter) use ($newValue) {
+
+            $asset = null;
+            if (is_array($newValue)) {
+                $asset = Asset::getById($newValue["id"]);
```
Comment (weisswurstkanone) [inline src/GraphQL/InputProcessor/Image.php]: Please compare with null or instance of Image
it is currently possible to assign any kind of asset

**Comment by weisswurstkanone** [comment]
please add it to the mutations.md doc

File: src/GraphQL/Type/InputType/AbstractRelationInputType.php (hunk 2 above)
Comment (dvesh3) [reply src/GraphQL/Type/InputType/AbstractRelationInputType.php]: so then user will have to provide ` image: {id:0} `to flush the image?

Comment (dvesh3) [reply src/GraphQL/Type/InputType/AbstractRelationInputType.php]: Added check in InputProcessor https://github.com/pimcore/data-hub/pull/70/files#diff-ce28ed7c2cab2b7a0025f433d8dede3fR37

**Pull request merged**